### ISSUE TITLE
[Bug] Date parsing error after installing v5.0.1 via conda-forge

### ISSUE THREAD

**Issue by asjohnston-asf** [asfadmin/Discovery-asf_search#124]
I receive a `regex._regex_core.error: bad escape \d at position 7` exception when searching using the `start` parameter to `asf_search.search` after installing asf_search v5.0.1 from conda forge:
```
(base) $ conda create --name foo asf_search=5.0.1
...
(base) $ conda activate foo
(foo) $ python
Python 3.10.5 | packaged by conda-forge | (main, Jun 14 2022, 07:04:59) [GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import asf_search
>>> asf_search.search(start='2022-07-20')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/asjohnston/miniconda3/envs/foo/lib/python3.10/site-packages/asf_search/search/search.py", line 94, in search
    kw_opts = ASFSearchOptions(**kwargs)
  File "/home/asjohnston/miniconda3/envs/foo/lib/python3.10/site-packages/asf_search/ASFSearchOptions/ASFSearchOptions.py", line 21, in __init__
    self.__setattr__(key, value)
  File "/home/asjohnston/miniconda3/envs/foo/lib/python3.10/site-packages/asf_search/ASFSearchOptions/ASFSearchOptions.py", line 39, in __setattr__
    super().__setattr__(key, validate(key, value))
  File "/home/asjohnston/miniconda3/envs/foo/lib/python3.10/site-packages/asf_search/ASFSearchOptions/validator_map.py", line 19, in validate
    return validator_map[key](value)
  File "/home/asjohnston/miniconda3/envs/foo/lib/python3.10/site-packages/asf_search/ASFSearchOptions/validators.py", line 52, in parse_date
    d = dateparser.parse(str(value))
  File "/home/asjohnston/miniconda3/envs/foo/lib/python3.10/site-packages/dateparser/conf.py", line 89, in wrapper
    return f(*args, **kwargs)
  File "/home/asjohnston/miniconda3/envs/foo/lib/python3.10/site-packages/dateparser/__init__.py", line 54, in parse
    data = parser.get_date_data(date_string, date_formats)
  File "/home/asjohnston/miniconda3/envs/foo/lib/python3.10/site-packages/dateparser/date.py", line 421, in get_date_data
    parsed_date = _DateLocaleParser.parse(
  File "/home/asjohnston/miniconda3/envs/foo/lib/python3.10/site-packages/dateparser/date.py", line 178, in parse
    return instance._parse()
  File "/home/asjohnston/miniconda3/envs/foo/lib/python3.10/site-packages/dateparser/date.py", line 182, in _parse
    date_data = self._parsers[parser_name]()
  File "/home/asjohnston/miniconda3/envs/foo/lib/python3.10/site-packages/dateparser/date.py", line 196, in _try_freshness_parser
    return freshness_date_parser.get_date_data(self._get_translated_date(), self._settings)
  File "/home/asjohnston/miniconda3/envs/foo/lib/python3.10/site-packages/dateparser/date.py", line 234, in _get_translated_date
    self._translated_date = self.locale.translate(
  File "/home/asjohnston/miniconda3/envs/foo/lib/python3.10/site-packages/dateparser/languages/locale.py", line 131, in translate
    relative_translations = self._get_relative_translations(settings=settings)
  File "/home/asjohnston/miniconda3/envs/foo/lib/python3.10/site-packages/dateparser/languages/locale.py", line 158, in _get_relative_translations
    self._generate_relative_translations(normalize=True))
  File "/home/asjohnston/miniconda3/envs/foo/lib/python3.10/site-packages/dateparser/languages/locale.py", line 172, in _generate_relative_translations
    pattern = DIGIT_GROUP_PATTERN.sub(r'?P<n>\d+', pattern)
  File "/home/asjohnston/miniconda3/envs/foo/lib/python3.10/site-packages/regex/regex.py", line 702, in _compile_replacement_helper
    is_group, items = _compile_replacement(source, pattern, is_unicode)
  File "/home/asjohnston/miniconda3/envs/foo/lib/python3.10/site-packages/regex/_regex_core.py", line 1737, in _compile_replacement
    raise error("bad escape \\%s" % ch, source.string, source.pos)
regex._regex_core.error: bad escape \d at position 7
```

The same `asf_search.search` call succeeds when installing v5.0.1 via pip:
```
(base) $ conda create --name bar pip
...
(base) $ conda activate bar
(bar) $ pip install asf_search==5.0.1
...
(bar) $ python
Python 3.10.5 | packaged by conda-forge | (main, Jun 14 2022, 07:06:46) [GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import asf_search
>>> results = asf_search.search(start='2022-07-20')
/home/asjohnston/miniconda3/envs/bar/lib/python3.10/site-packages/dateparser/date_parser.py:35: PytzUsageWarning: The localize method is no longer necessary, as this time zone supports the fold attribute (PEP 495). For more details on migrating to a PEP 495-compliant implementation, see https://pytz-deprecation-shim.readthedocs.io/en/latest/migration.html
  date_obj = stz.localize(date_obj)
>>> len(results)
2041
```

**Comment by jhkennedy** [comment]
I'm able to reproduce this and it's likely related to this:
https://stackoverflow.com/questions/71496687/dateparser-throws-regex-regex-core-error

**Comment by jhkennedy** [comment]
Okay, so, I know what's happening now, but not sure yet how to fix. The rundown:

* `dateparser` pins `regex` to [`regex !=2019.02.19,!=2021.8.27,<2022.3.15`](https://github.com/scrapinghub/dateparser/blob/master/setup.py#L30)
* that is reflected in the `dateparser` conda build feedstock: https://github.com/conda-forge/dateparser-feedstock/blob/main/recipe/meta.yaml#L26
  * Note: current version would be `v1.1.1`
* and `asf_search` doesn't pin the `dateparser` version number

But, in my conda environment, I end up with `dateparser=1.0.0`:
```
$ conda create -n foo asf_search
$ conda activate foo
$ conda list | grep dateparser
dateparser                1.0.0              pyhd3deb0d_0    conda-forge
```
Looking at `dateparser` v1.0.0's build recipe:
https://github.com/conda-forge/dateparser-feedstock/blob/84344d7a9a71c00ffd56f6665dd602536ec57888/recipe/meta.yaml#L24
there is no `regex` top pin, so we end up just getting the latest `regex`. 

So something in the dependency chain is restricting us to `dataparser` 1.0.0... that'll take some tracking.

**Comment by jhkennedy** [comment]
This is a conda-forge dependency issue, so I'm going to move this issue to the conda-forge feedstock for `asf_search` and continue the discussion there.

**Comment by jhkennedy** [comment]
Nevermind I can't transfer it there. harumph.

**Comment by jhkennedy** [comment]
Okay, more fun -- if you use `mamba` instead of `conda` it works correctly. So this a "weird" conda bug. I think the "solution" would be to just put a bottom pin on `dateparser` in the conda build recipe *even though we shouldn't need it.*

**Comment by jhkennedy** [comment]
Fixed in conda-forge/asf_search-feedstock#32 for `asf_search` v5.0.2 and greater. If you encounter this with a previous version, consider upgrading, or using `mamba` instead.
